File: app/src/main/java/com/task/ui/component/details/DetailsContract.java
package com.task.ui.component.details;

import com.task.data.remote.dto.NewsItem;
import com.task.ui.base.listeners.BaseView;

/**
 * Created by AhmedEltaher on 11/12/16.
 */

public interface DetailsContract extends BaseView {
    interface View extends BaseView {
        void initializeView(NewsItem newsItem);
    }

    interface Presenter {
        String getMainImageURL();
    }
}

File: app/src/main/java/com/task/ui/component/details/DetailsContract.kt
package com.task.ui.component.details

import com.task.data.remote.dto.NewsItem
import com.task.ui.base.listeners.BaseView

/**
 * Created by AhmedEltaher on 11/12/16.
 */

interface DetailsContract : BaseView {
    interface View : BaseView {
        fun initializeView(newsItem: NewsItem)
    }

    interface Presenter
}
